File: Desafios Orange/orange-talents-05-template-casa-do-codigo/casa-do-codigo/src/main/java/br/com/zup/casadocodigo/casadocodigo/repositorios/AutorRepository.java
package br.com.zup.casadocodigo.casadocodigo.repositorios;

import br.com.zup.casadocodigo.casadocodigo.entidades.Autor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface AutorRepository extends JpaRepository<Autor, Long> {

    public Optional<Page<Autor>> findByEmail(String email, Pageable paginacao);

    public Optional<Autor> findByEmail(String email);



}

File: Cursos/Curso 17 - Micronaut/src/main/kotlin/br/com/zup/novoautor/AutorRepository.kt
package br.com.zup.novoautor

import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaRepository
import java.util.*

@Repository
interface AutorRepository: JpaRepository<Autor, Long> {

    fun findByEmail(email: String) : Optional<Autor>
}